query-engine/query-engine/src/capture_tracer.rs

```diff
@@ -0,0 +1,108 @@
+use async_trait::async_trait;
+use opentelemetry::{
+    global,
+    sdk::{self},
+    sdk::{
+        export::trace::{ExportResult, SpanData, SpanExporter},
+        propagation::TraceContextPropagator,
+    },
+    trace::TracerProvider,
+};
+use query_core::spans_to_json;
+use std::fmt::Debug;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::Mutex;
+
+/// Pipeline builder
+#[derive(Debug)]
+pub struct PipelineBuilder {
+    trace_config: Option<sdk::trace::Config>,
+}
+
+/// Create a new stdout exporter pipeline builder.
+pub fn new_pipeline() -> PipelineBuilder {
+    PipelineBuilder::default()
+}
+
+impl Default for PipelineBuilder {
+    /// Return the default pipeline builder.
+    fn default() -> Self {
+        Self { trace_config: None }
+    }
+}
+
+impl PipelineBuilder {
+    /// Assign the SDK trace configuration.
+    pub fn with_trace_config(mut self, config: sdk::trace::Config) -> Self {
+        self.trace_config = Some(config);
+        self
+    }
+}
+
+impl PipelineBuilder {
+    pub fn install_simple(mut self, exporter: CaptureExporter) -> sdk::trace::Tracer {
+        global::set_text_map_propagator(TraceContextPropagator::new());
+
+        let mut provider_builder = sdk::trace::TracerProvider::builder().with_simple_exporter(exporter);
+        if let Some(config) = self.trace_config.take() {
+            provider_builder = provider_builder.with_config(config);
+        }
+        let provider = provider_builder.build();
+        let tracer = provider.tracer("opentelemetry");
+        global::set_tracer_provider(provider);
+
+        tracer
+    }
+}
+
+/// A [`CaptureExporter`] that sends spans to stdout.
+#[derive(Debug, Clone)]
+pub struct CaptureExporter {
+    logs: Arc<Mutex<HashMap<String, Vec<SpanData>>>>,
```

Comment:
Sorry if this is an irrelevant question, I'm missing a lot of context here anyway, but I was curious looking at the code why not use `opentelemetry::trace::TraceId` here and in `struct LogCapture` directly and avoid converting it to string everywhere.

Reply:
Good point. We can change it to that